x-pack/plugins/triggers_actions_ui/public/application/hooks/use_load_rule_aggregations.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { useState, useCallback } from 'react';
+import { RuleExecutionStatusValues } from '@kbn/alerting-plugin/common';
+import { loadRuleAggregations, LoadRuleAggregationsProps } from '../lib/rule_api';
+import { useKibana } from '../../common/lib/kibana';
+
+type UseLoadRuleAggregationsProps = Omit<LoadRuleAggregationsProps, 'http'> & {
+  onError: (message: string) => void;
+};
+
+export function useLoadRuleAggregations({
+  searchText,
+  typesFilter,
+  actionTypesFilter,
+  ruleExecutionStatusesFilter,
+  ruleStatusesFilter,
+  tagsFilter,
+  onError,
+}: UseLoadRuleAggregationsProps) {
+  const { http } = useKibana().services;
+
+  const [rulesStatusesTotal, setRulesStatusesTotal] = useState<Record<string, number>>(
+    RuleExecutionStatusValues.reduce(
+      (prev: Record<string, number>, status: string) =>
+        ({
+          ...prev,
+          [status]: 0,
+        } as Record<string, number>),
+      {}
+    )
+  );
+
+  const internalLoadRuleAggregations = useCallback(async () => {
+    try {
+      const rulesAggs = await loadRuleAggregations({
+        http,
+        searchText,
+        typesFilter,
+        actionTypesFilter,
+        ruleExecutionStatusesFilter,
+        ruleStatusesFilter,
+        tagsFilter,
+      });
+      if (rulesAggs?.ruleExecutionStatus) {
+        setRulesStatusesTotal(rulesAggs.ruleExecutionStatus);
+      }
+    } catch (e) {
+      onError(
+        i18n.translate(
+          'xpack.triggersActionsUI.sections.rulesList.unableToLoadRuleStatusInfoMessage',
+          {
+            defaultMessage: 'Unable to load rule status info',
+          }
+        )
+      );
+    }
+  }, [
+    http,
+    searchText,
+    typesFilter,
+    actionTypesFilter,
+    ruleExecutionStatusesFilter,
+    ruleStatusesFilter,
+    tagsFilter,
+    onError,
+    setRulesStatusesTotal,
+  ]);
+
+  return {
```

Comment:
I think we should memoized the return if not it might have some re-rendering, what do you think?

Reply:
I dont mind, although the references for all of the properties within the return object do not change. Since we end up destructuring in the component that uses it